LuaMenu/widgets/chobby/components/ai_list_window.lua

```diff
@@ -70,42 +72,88 @@ function AiListWindow:AddAiToList(ai, blackList, oldAiVersions, isRunning64Bit)
 	if Configuration.gameConfig.aiTooltip then
 		tooltip = Configuration.gameConfig.aiTooltip[displayName]
 	end
+
+	local buttonList = nil
+	if Configuration.showAiOptions then
+		local path = "AI/Skirmish/" .. shortName .. "/" .. ai.version .. "/AIOptions.lua"
+		if VFS.FileExists(path) then
+			buttonList = self:MakeAiOptionsButton(displayName, tooltip, shortName, ai.version, path)
+		end
+	end
+	if buttonList == nil then
+		buttonList = self:MakeAiButton(displayName, tooltip, shortName, ai.version)
+	end
+	self:AddRow(buttonList, displayName)
+end
+
+function AiListWindow:MakeAiOptionsButton(displayName, tooltip, shortName, version, path)
 	local addAIButton = Button:New {
 		x = 0,
 		y = 0,
-		width = "100%",
+		width = "80%",
 		height = "100%",
 		caption = displayName,
 		font = Configuration:GetFont(3),
 		tooltip = tooltip,
 		OnClick = {
 			function()
-				local function defaultAction()
-					self:AddAi(displayName, shortName, ai.version)
+				self:AddAi(displayName, shortName, version)
+				self:HideWindow()
+			end
+		},
+	}
+	local optionsButton = Button:New {
+		x = "80%",
+		y = 0,
+		width = "20%",
+		height = "100%",
+		caption = "",
+		font = Configuration:GetFont(3),
+		OnClick = {
+			function()
+				if self.lobby.name ~= "singleplayer" then
```

Comment:
Here's what created a decision issue for me: not showing AIOptions button in MP would be best (workaround so far). But because ai_list_window is cached for performance and the list with buttons are created on init  self.lobby is not initialized. I'll add re-building ai list on every self.lobby change (requires passing battleLobby param on init). The reason why i didn't want to do it is that implementing protocol command for MP is the proper way to go (and after that lobby param will have to be removed again, to reduce unnecessary ai list re-builds). Ok, do not merge it yet.

Reply:
Sone updating